hw/drivers/lp5523/src/lp5523.c

```diff
@@ -0,0 +1,1144 @@
+    return 0;
+}
+
+/**
+ * Reads a section from the specified register
+ *
+ * @param The sensor interface
+ * @param The register address to read from
+ * @param The mask
+ * @param Pointer to where the value should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+lp5523_get_val(struct sensor_itf *itf, enum lp5523_registers addr,
+    uint8_t mask, uint8_t *value)
+{
+    int rc;
+    uint8_t reg;
+
+    uint8_t shift;
+
+    shift = lp5523_clz(mask);
+
+    rc = lp5523_get_reg(itf, addr, &reg);
+
+    *value = (reg & mask) >> shift;
+
+    return rc;
+}
+
+/**
+ * Writes 9 bits to 2 registers, useful to write to:
+ * LP5523_BF_OUTPUT_RATIOMETRIC
+ * LP5523_BF_OUTPUT_CONTROL
+ * LP5523_BF_ENG_MAPPING (see lp5523_set_engine_mapping())
+ *
+ * @param The sensor interface
+ * @param The register address to write to (the MSB register)
+ * @param The values to write, only the least significant 9 bits are significant
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+
+int
+lp5523_set_bitfield(struct sensor_itf *itf, enum lp5523_bitfield_registers addr,
+    uint16_t ops)
+{
+    uint8_t vals[2] = { (ops >> 8) & 0x01, ops & 0xff };
+
+    return lp5523_set_2_regs(itf, addr, vals);
+}
+
+/**
+ * Writes 9 bits to 2 registers, useful to read from:
+ * LP5523_BF_OUTPUT_RATIOMETRIC
+ * LP5523_BF_OUTPUT_CONTROL
+ * LP5523_BF_ENG_MAPPING (see lp5523_set_engine_mapping())
+ *
+ * @param The sensor interface
+ * @param The register address to read from (the MSB register)
+ * @param Pointer to where the register values should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+
+int
+lp5523_get_bitfield(struct sensor_itf *itf, enum lp5523_bitfield_registers addr,
+    uint16_t* ops)
+{
+    int rc;
+    uint8_t vals[2];
+
+    rc = lp5523_get_2_regs(itf, addr, vals);
+
+    *ops = ((vals[0] & 0x01) << 8) | vals[1];
+
+    return rc;
+}
+
+/**
+ * Writes to a register with an output-based address offset,
+ * useful to write to:
+ * LP5523_OP_CONTROL
+ * LP5523_OP_PWM
+ * LP5523_OP_CURRENT_CONTROL
+ *
+ * @param The sensor interface
+ * @param The register address to write to (the MSB register)
+ * @param Value to write
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+lp5523_set_output_reg(struct sensor_itf *itf, enum lp5523_output_registers addr,
+    uint8_t op, uint8_t value)
+{
+    --op;
+    return (op > 8) ? -1 : lp5523_set_reg(itf, addr + op, value);
+}
+
+/**
+ * Reads from a register with an output-based address offset,
+ * useful to read from:
+ * LP5523_OP_CONTROL
+ * LP5523_OP_PWM
+ * LP5523_OP_CURRENT_CONTROL
+ *
+ * @param The sensor interface
+ * @param The register address to read from (the MSB register)
+ * @param Pointer to where the register values should be written
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+lp5523_get_outout_reg(struct sensor_itf *itf, enum lp5523_output_registers addr,
+    uint8_t op, uint8_t *value)
+{
+    --op;
```

Comment:
One argument is not commented for here: "op". What if op is 0 ? Is that possible ?